lib/zone.js

```diff
@@ -1,14 +1,17 @@
 'use strict';
 
 var core = require('./core');
-var browserPatch = require('./patch/browser');
+var nodePatch = require('./patch/node');
+var longStackTraceZone = require('./zones/long-stack-trace');
 
-global.zone = new core.Zone();
+if (!global.zone) {
+  global.zone = new core.Zone();
+  core.Zone.longStackTraceZone = longStackTraceZone;
+
+  nodePatch.apply();
+}
 
 module.exports = {
```

Comment:
It would be nice if the exports here were `rootZone`, `bindPromiseFn`, and `longStackTraceZone`. Then there'd be no need to put them on the `Zone` ctor on node. But I imagine having the public APIs roughly equal for node and browser is important?